From 71a0056deaf906151c036c378b4353a6c67593c0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 15 Jun 2021 22:05:08 +0300
Subject: [PATCH] [SW-45909] fwupd: on "run" close both memory areas

Current code executes FIT according to data received in MAP, but toggles
the security closure according to the RUN command. This enables a
MAP_SRAM -> RUN_SPSRAM attack surface when SRAM is written, but SPSRAM
is closed during validation.
When receiving HL_UPD_RUN_*SRAM command, close both memory areas to
prevent MAP_SRAM -> RUN_SPSRAM attack surface.

Change-Id: Ib5befa55049856e8fdd98cb9700cb5c6dab1b6f9
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index f8631f2..afedbe9 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -619,10 +619,12 @@
 	int err;
 	enum hl_upd_sts sts = HL_UPD_STS_OK;
 
-	if (cmd == HL_UPD_RUN_SRAM)
-		preboot_security_dynamic_img_sram_close();
-	else
-		preboot_security_dynamic_img_spsram_close();
+	/*
+	 * we do not allow mapping of SRAM and SPSRAM at the same time -
+	 * close both.
+	 */
+	preboot_security_dynamic_img_sram_close();
+	preboot_security_dynamic_img_spsram_close();
 
 	err = module_fwupd_load_type(state);
 
